pyiron_base/jobs/job/extension/server/generic.py

```diff
@@ -441,6 +454,35 @@ def view_queues():
         else:
             return None
 
+    @property
+    def executor(self) -> Union[Executor, None]:
+        if self.run_mode.executor:
+            return self._executor
+        else:
+            self._executor = None
+            return self._executor
```

Comment:
```suggestion
        if not self.run_mode.executor and self._executor is not None:
            self._executor = None
        return self._executor
```

Cleaner with a single return loop, IMO